hooks/use-sound.ts

```diff
@@ -0,0 +1,103 @@
+'use client';
+import type {State} from '@app/config/state';
+import {useCallback, useContext, useEffect, useState} from 'react';
+import {Context} from '../contexts/audio';
+
+const useSound = (state: State): void => {
+	const [loadedSounds, setLoadedSounds] = useState<{[url: string]: AudioBuffer}>({});
+	const [lastTimestamp, setLastTimestamp] = useState<number>();
+	const audioContext = useContext(Context);
+
+	const loadSound = async (url: string): Promise<AudioBuffer> => {
+		// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+		if (loadedSounds[url] !== undefined) {
+			return loadedSounds[url];
+		}
+
+		if (audioContext === undefined) {
+			throw new Error('Audio context is undefined');
+		}
+
+		const response = await fetch(url);
+		const arrayBuffer = await response.arrayBuffer();
+
+		const sound = await audioContext.decodeAudioData(arrayBuffer);
+
+		setLoadedSounds({
+			...loadedSounds,
+			[url]: sound,
+		});
+		
+		return sound;
+	};
+
+	const playSound = useCallback(
+		// eslint-disable-next-line unicorn/no-object-as-default-parameter
+		async (url: string, options = {detune: 0}): Promise<void> => {
+			if (audioContext === undefined) {
+				return;
+			}
+			
+			const sound = await loadSound(url);
+
+			const source = audioContext.createBufferSource();
+			source.buffer = sound;
+
+			source.detune.value = options.detune;
+
+			source.connect(audioContext.destination);
+			source.start(0);
+		}, [audioContext, loadedSounds],
+	);
+
+	useEffect(() => {
+		if (state.enableSound === false || lastTimestamp === state.lastEventTime || state.lastEvent === undefined) {
+			return;
+		}
+
+		let sound = '';
+		let detune = 0;
+
+		const sounds = state.sounds;
+
+		switch (state.lastEvent) {
+			case 'type': {
+				sound = sounds.type[Math.floor(Math.random() * sounds.type.length)];
+				break;
+			}
+			case 'wordComplete': {
+				sound = sounds.wordComplete[Math.floor(Math.random() * sounds.gameComplete.length)];
+				detune = 100 + 100 * (state.word.streak / state.targetStreak); 
+				break;
+			}
+			case 'streakComplete': {
+				sound = sounds.streakComplete[Math.floor(Math.random() * sounds.streakComplete.length)];
+				break;
+			}
+			case 'failureSlow': {
+				sound = sounds.failureSlow[Math.floor(Math.random() * sounds.failureSlow.length)];
+				break;
+			}
+			case 'failureTypo': {
+				if (!(state.word.characters[0].correct ?? false)) {
+					return;
+				} 
```

Comment:
Can you explain why this conditional is required?

Reply:
It's not necessarily required but I felt it was nicer when the error sound only appeared when you've actually started typing the word. This just checks that you're not on the first character - happy to let that go though

Reply:
I suggest we remove it, as you're stopping the audio feedback you'd typically hear if you incorrectly type the first character of a word, which may be confusing since you're conditioning users to expect a specific sound on a typo. Without that happening on the first letter of a word, you might not realise you're making a typo and instead think the app or your keyboard isn't working correctly :)